core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/DetailedSampleBuilder.java

```diff
@@ -875,23 +871,24 @@ public DetailedSample build() {
 
     if (parent != null) {
       sample.setParent(parent);
-    } else if (!Identity.CATEGORY_NAME.equals(sampleClass.getSampleCategory())) {
-      Identity identity = new IdentityImpl();
-      if (externalName == null) {
-        throw new NullPointerException("Missing externalName");
+    } else {
+      DetailedSample parent = null;
+      if (!Identity.CATEGORY_NAME.equals(sampleClass.getSampleCategory()) && externalName != null) {
+        parent = buildIdentity();
       }
-      identity.setExternalName(externalName);
-      identity.setInternalName(internalName);
-      identity.setDonorSex(donorSex);
-      if (SampleTissue.CATEGORY_NAME.equals(sampleClass.getSampleCategory())) {
-        sample.setParent(identity);
-      } else {
-        if (tissueClass == null) throw new NullPointerException("Missing tissue class");
+      if (!SampleTissue.CATEGORY_NAME.equals(sampleClass.getSampleCategory()) && tissueClass != null) {
         SampleTissue tissue = buildTissue();
+        tissue.setParent(parent);
         tissue.setSampleClass(tissueClass);
-        tissue.setParent(identity);
-        sample.setParent(tissue);
+        parent = tissue;
+      }
+      if (!SampleStock.CATEGORY_NAME.equals(sampleClass.getSampleCategory())) {
```

Comment:
If the sample being built is a Tissue or Identity, won't this result in a Stock being made as parent of the Tissue?